src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.List;
+import mate.academy.spring.dao.UserDao;
+import mate.academy.spring.model.User;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void save(User user) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save product to DB", e);
+        }
```

Comment:
if something does wrong - we won't be able to rollback transaction - because session is already closed
use try-catch-finally

Reply:
Done